winrm/command_test.go

```
@@ -75,3 +75,70 @@ func (s *WinRMSuite) TestStdinCommand(c *C) {
 	go io.Copy(&errWriter, command.Stderr)
 	command.Wait()
 }
+
+func (s *WinRMSuite) TestExecuteCommandCertAuth(c *C) {
+	client, err := NewClient(&Endpoint{Host: "localhost", Port: 5986, HTTPS: true, Cert: []byte(cert), Key: []byte(key)}, "", "")
+	c.Assert(err, IsNil)
+
+	shell := &Shell{client: client, ShellId: "67A74734-DD32-4F10-89DE-49A060483810"}
+	count := 0
+	client.http = func(client *Client, message *soap.SoapMessage) (string, error) {
+		switch count {
+		case 0:
+			{
+				c.Assert(message.String(), Contains, "http://schemas.microsoft.com/wbem/wsman/1/windows/shell/Command")
+				count = 1
+				return executeCommandResponse, nil
+			}
+		case 1:
+			{
+				c.Assert(message.String(), Contains, "http://schemas.microsoft.com/wbem/wsman/1/windows/shell/Receive")
+				count = 2
+				return outputResponse, nil
+			}
+		default:
+			{
+				return doneCommandResponse, nil
+			}
+		}
+	}
+
+	command, _ := shell.Execute("ipconfig /all")
```

Can this error not be swallowed and tested?